### PR TITLE
Fixed parsing of TECH10 output in newer Mplus versions

### PR DESCRIPTION
**Pull request by sam-crawley**
An additional line was added, so updated the parsing to account for this, and added a test.

The first commit is a simple fix for the bug, the second reworks TECH10 parsing completely, and also extracts the new value in newer versions of Mplus.

### CODE REVIEW

**Comment by sam-crawley** [comment]
This fixes #173

File: DESCRIPTION [hunk 1]
```diff
@@ -41,7 +41,8 @@ Imports:
     data.table,
     rlang,
     fastDummies,
-    checkmate
+    checkmate,
+    reshape2
```
Comment (cjvanlissa) [inline DESCRIPTION:45]: Could you use base R reshape instead of incurring a dependency on reshape2?

File: R/parseOutput.R [hunk 2]
```diff
@@ -1845,58 +1845,49 @@ extractTech10 <- function(outfiletext, filename) {
...
   
-  bivarFitStats <- setNames(data.frame(bivarFitStats, stringsAsFactors = FALSE), c("var1","var2","pearson","log-likelihood"))
-  bivarFitStats[,c("pearson","log-likelihood")] <- as.numeric(unlist(bivarFitStats[,c("pearson","log-likelihood")]))
+  names(bivarFitStats) <- c("var1", "var2", "stat", "value")
+  bivarFitStats <- dcast(bivarFitStats, var1 + var2 ~ stat)
   
   tech10List$bivar_model_fit_info <- bivarFitData
-  tech10List$bivar_chi_square <- bivarFitStats
-  
+  tech10List$bivar_stats <- bivarFitStats
```
Comment (cjvanlissa) [inline R/parseOutput.R:1890]: Any concern about breaking backwards compatibility of MplusAutomation by changing the name of this output element from bivar_chi_square to bivar_stats?

**Comment by cjvanlissa** [comment]
Thank you for doing this, @sam-crawley ! Including tests was a wonderful touch. Just wondering if the function also still works with old-style output?

File: R/parseOutput.R (hunk 2 above)
Comment (sam-crawley) [reply R/parseOutput.R:1898]: Done

File: DESCRIPTION (hunk 1 above)
Comment (sam-crawley) [reply DESCRIPTION]: Done

File: R/parseOutput.R (hunk 2 above)
Comment (sam-crawley) [reply R/parseOutput.R:1898]: Yes, it's backwards incompatible, but it's an easy fix and the new name is a better description of what it contains. But happy to change it back if you'd prefer.

**Comment by sam-crawley** [comment]
I've added a test with an output generated from an older Mplus version.

File: R/parseOutput.R (hunk 2 above)
Comment (cjvanlissa) [reply R/parseOutput.R:1898]: From a reproducibility perspective, it's important that the name stays the same. There's also CRAN policy about anouncing breaking changes one major update ahead of time with a warning.. let's avoid all that and just keep the name the same.

**Comment by cjvanlissa** [comment]
Thank you very much @sam-crawley!

**Comment by michaelhallquist** [comment]
Indeed, thanks for this update, @sam-crawley! And thanks to @cjvanlissa for merging it into the codebase. I just reviewed the code as well and it looks great to me. Next time I roll out a CRAN update, I'll put this code through my internal checks, which has TECH10 outputs going back to Mplus 5. But I appreciate your adding formal tests for the update, which is definitely the direction things should go.

Cheers,
Michael

**Pull request merged**